src/EncompassRest/ResourceLocks/ResourceLocks.cs

```diff
@@ -0,0 +1,137 @@
+            LockResourceAsync(lockType, resourceId, resourceType.ToString(), force, populate, cancellationToken);
+        public Task<string> LockResourceAsync(ResourceLockType lockType, string resourceId, string resourceType, bool force, bool populate, CancellationToken cancellationToken = default) =>
+            LockResourceAsync(lockType.ToString(), resourceId, resourceType, force, populate, cancellationToken);
+        public Task<string> LockResourceAsync(string lockType, string resourceId, string resourceType, bool force, bool populate, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceId, nameof(resourceId));
+            Preconditions.NotNullOrEmpty(resourceType, nameof(resourceType));
+            Preconditions.NotNullOrEmpty(lockType, nameof(lockType));
+
+            ResourceLock resourceLock = new ResourceLock();
+            resourceLock.Resource.EntityId = resourceId;
+            resourceLock.Resource.EntityType = resourceType;
+            resourceLock.LockType = lockType;
+
+            return LockResourceAsync(resourceLock, force, populate, cancellationToken);
+        }
+
+        public Task<string> LockResourceRawAsync(string resource, string queryString, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resource, nameof(resource));
+
+            return PostAsync(null, queryString, new JsonStringContent(resource), nameof(LockResourceRawAsync), null, cancellationToken, ReadAsStringElseLocationFunc);
+        }
+        public Task<bool> UnlockResourceAsync(string lockId, string resourceId, EntityType resourceType, bool force, CancellationToken cancellationToken = default) =>
+            UnlockResourceAsync(lockId, resourceId, resourceType.ToString(), force, cancellationToken);
+        public Task<bool> UnlockResourceAsync(string lockId, string resourceId, string resourceType, bool force, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceId, nameof(resourceId));
+            Preconditions.NotNullOrEmpty(resourceType, nameof(resourceType));
+            Preconditions.NotNullOrEmpty(lockId, nameof(lockId));
+
+            var queryParameters = new QueryParameters();
+            queryParameters.Add(new QueryParameter("resourceType", resourceType));
+            queryParameters.Add(new QueryParameter("resourceId", resourceId));
+            if (force) queryParameters.Add(new QueryParameter("force", "true"));
+
+            return DeleteAsync(lockId, queryParameters.ToString(), cancellationToken);
+        }
+        public Task<bool> UnlockResourceAsync(ResourceLock resourceLock, CancellationToken cancellationToken = default) => UnlockResourceAsync(resourceLock, false, cancellationToken);
```

Comment:
Can we remove this `ResourceLock` input parameter override?

Reply:
it may be simpler to pass the ResourceLock object to the unlock method. it is a sugar, but i know i will be using it.